packages/sitecore-jss/src/native-fetcher.ts

```diff
@@ -123,4 +207,44 @@ export class NativeDataFetcher {
 
     return headers;
   }
+
+  /**
+   * Parses the response data.
+   * @param {Response} response - The fetch response object.
+   * @param {Function} debug - The debug logger function.
+   * @returns {Promise<unknown>} - The parsed response data.
+   */
+  private async parseResponse(
+    response: Response,
+    debug: (message: string, ...optionalParams: any[]) => void
+  ): Promise<unknown> {
+    const contentType = response.headers.get('Content-Type') || '';
+    try {
+      if (contentType.includes('application/json')) {
+        return await response.json();
+      }
+      return await response.text();
+    } catch (error) {
+      debug('Response parsing error: %o', error);
+      return undefined;
+    }
+  }
+
+  /**
+   * Creates a custom error for fetch failures.
+   * @param {Response} response - The fetch response object.
+   * @param {unknown} data - The parsed response data.
+   * @returns {NativeDataFetcherError} - The constructed error object.
+   */
+  private createError(response: Response, data: unknown): NativeDataFetcherError {
```

Comment:
Should `data` be optional?
```suggestion
  private createError(response: Response, data?: unknown): NativeDataFetcherError {
```